src/rust/util/src/decimal.rs

```diff
@@ -0,0 +1,51 @@
+// Copyright 2020 Shift Crypto AG
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+extern crate alloc;
+use alloc::string::String;
+
+/// Formats integer `value` as `value / 10^decimals`, with up to `decimals` decimal places.
+/// E.g. "123450" wit decimals=3: "123.45".
+/// Value must consists only if '0'-'9' digits.
+pub fn format(value: &str, decimals: usize) -> String {
```

Comment:
I'd actually take u64 here and do `satoshi.to_string()`. It just doesn't feel right - formatting a string with decimals which makes sense only for integers. Then, `format_amount` becomes:

```rust
    let mut s = util::decimal::format(satoshi, 8);
    s.push(' ');
    s.push_str(unit);
    s
```

You could even accept the unit here:

```rust
pub fn format(value: u64, decimals: usize, unit: &str) -> String
```

Reply:
see the commit message - this will also be used with ETH values that exceed u64.

for more type safety i could make a trait and put it on both u64 and BigUint? :thinking:

Reply:
I did it with a type constraint that I can also put on BigUint later :+1: